### PR TITLE
141 temporal key

### PR DESCRIPTION
**Pull request by idruzhitskiy**

### CODE REVIEW

**Comment by idruzhitskiy** [comment]
closes #143

File: src/blob/index/bptree/node.rs [hunk 1]
```diff
@@ -38,15 +38,18 @@ impl Node {
...
+    {
         let key_size = K::LEN as usize;
         let mut l = 0i32;
         let mut r: i32 = (buf.len() / key_size - 1) as i32;
         while l <= r {
             let m = (l + r) / 2;
             let offset = m as usize * key_size;
-            let k = buf[offset..(offset + key_size)].to_vec().into(); // TODO Allow keys to be based on slices
-            match key.cmp(&k) {
+            let k = K::Ref::from(&buf[offset..(offset + key_size)]); // TODO Allow keys to be based on slices
```
Comment (ikopylov) [inline src/blob/index/bptree/node.rs:51]: `TODO` can be removed

**Comment by ikopylov** [comment]
Please, fix build errors, that occur after https://github.com/qoollo/pearl/pull/103 was merged

File: src/storage/core.rs [hunk 2]
```diff
@@ -1048,18 +1066,33 @@ impl<K: Key + 'static> Safe<K> {
...
     }
+
+    /// Convert `Self` to `Self::Ref`
+    fn as_ref_key(&'a self) -> Self::Ref {
+        Self::Ref::from(self.as_ref())
+    }
 }
 
+/// Trait for reference key type
+pub trait RefKey<'a>: Ord + From<&'a [u8]> {}
```
Comment (ikopylov) [inline src/storage/core.rs:1090]: Please add an empty line after the `RefKey` declaration to visually separate it from `BloomProvider`

File: src/blob/index/bptree/node.rs (hunk 1 above)
Comment (idruzhitskiy) [reply src/blob/index/bptree/node.rs:51]: Done

File: src/storage/core.rs (hunk 2 above)
Comment (idruzhitskiy) [reply src/storage/core.rs:1090]: Done

**Review by ikopylov — APPROVED** [review]
LGTM

**Pull request merged**